app/src/main/java/com/example/playlistmaker/SearchActivity.kt

```diff
@@ -1,20 +1,88 @@
 package com.example.playlistmaker
 
+import android.content.Context
 import android.os.Bundle
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.View
+import android.view.inputmethod.InputMethodManager
+import android.widget.EditText
+import android.widget.ImageButton
+import android.widget.ImageView
 import androidx.activity.enableEdgeToEdge
 import androidx.appcompat.app.AppCompatActivity
 import androidx.core.view.ViewCompat
 import androidx.core.view.WindowInsetsCompat
 
 class SearchActivity : AppCompatActivity() {
+
+    private lateinit var searchEditText: EditText
+    private lateinit var clearSearchButton: ImageView
+    private var searchQuery: String = ""  // Переменная для сохранения текста поиска
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         enableEdgeToEdge()
         setContentView(R.layout.activity_search)
+
+        // Настройка отступов для системных баров
         ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
             val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
             v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
             insets
         }
+
+        val buttonBack = findViewById<ImageButton>(R.id.backSearch)
+        buttonBack.setOnClickListener { finish() }
+
+        searchEditText = findViewById(R.id.searchView)
+        clearSearchButton = findViewById(R.id.clearIcon)
+
+        // Установка сохранённого текста (если экран перевернули)
+        searchEditText.setText(searchQuery)
+
+        searchEditText.addTextChangedListener(object : TextWatcher {
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
+
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                searchQuery = s.toString()  // Сохраняем введённый текст
+                clearSearchButton.visibility = if (s.isNullOrEmpty()) View.GONE else View.VISIBLE
+            }
+
+            override fun afterTextChanged(s: Editable?) {}
+        })
+
+        clearSearchButton.setOnClickListener {
+            searchEditText.text.clear()
+            searchEditText.requestFocus()
+            showKeyboard(searchEditText)
```

Comment:
Здесь видимо опечатка, т.к. нужно скрывать клавиатуру:
```suggestion
            searchEditText.clearFocus()
            hideKeyboard()
```